lib/Frontend/nvqpp/ASTBridge.cpp

```diff
@@ -342,6 +342,24 @@ ASTBridgeAction::ASTBridgeConsumer::ASTBridgeConsumer(
   assert(mangler && "mangler creation failed");
 }
 
+void ASTBridgeAction::ASTBridgeConsumer::addFunctionDecl(
+    const clang::FunctionDecl *funcDecl, details::QuakeBridgeVisitor &visitor,
+    FunctionType funcTy) {
+  auto funcName = visitor.cxxMangledDeclName(funcDecl);
+  if (module->lookupSymbol(funcName))
+    return;
+  auto loc = toSourceLocation(module->getContext(), &astContext,
+                              funcDecl->getSourceRange());
+  OpBuilder build(module->getBodyRegion());
+  OpBuilder::InsertionGuard guard(build);
+  build.setInsertionPointToEnd(module->getBody());
+  if (isa<clang::CXXMethodDecl>(funcDecl))
+     funcTy = cudaq::opt::factory::toCpuSideFuncType(funcTy);
+  auto func = build.create<func::FuncOp>(loc, funcName, funcTy,
```

Comment:
since we add `cudaq-entrypoint` to the main function, I think we should add `cpp-entrypoint` to this function, just so you're not left wondering what this function declaration is for when you output the quake.

Reply:
This declaration is transient though. It'll be replaced when the GenKernelExecution pass runs. So typically a user isn't even going to see it.

Reply:
I was just thinking about someone who ran cudaq-quake (like me 😄 ). This is not a deal breaker for me or something to hold this up. So no worries.